Bluetooth: Implement Secondary_Advertising_Max_Skip in controller

Tronil opened the pull request:
Fix the host side not setting sec_adv_max_skip

Behaviour is unchanged for max_skip == 0

For max_skip > 0, use a slightly different calculation to ensure we can pass LL/DDI/ADV/BV-28-C

cvinayak commented on subsys/bluetooth/controller/Kconfig.ll_sw_split:
```diff
@@ -335,6 +335,15 @@ config BT_CTLR_ADV_AUX_SET
 	help
 	  Maximum supported advertising auxiliary channel sets.
 
+config BT_CTLR_ADV_USE_MAX_SECONDARY_INTERVAL
+	bool "Use the maximum possible interval between ADV_AUX_INDs"
+	depends on BT_BROADCASTER && BT_CTLR_ADV_EXT
+	default n
+	help
+	  Use the value of Secondary_Advertising_Max_Skip to only send
+	  ADV_AUX_IND when required - ie. skip as many as allowed by the
+	  parameters set by the host.
+
```
If this is not something configurable in IXIT or ICS, lets not have a Kconfig (in this PR Host will always provide a valid value anyway).

Tronil replied:
True, that would be preferable; However, this has some implications for qualification (specifically LL/DDI/ADV/BV-28-C). I'll see if I can get some time to look at it next week, but otherwise it will have to wait until I get back from my summer vacation (ie. late August).

Tronil replied:
Changed the implementation to avoid the configuration flag; This means using a slightly different calculation for max_skip == 0 to avoid a change in behaviour while still passing LL/DDI/ADV/BV-28-C for higher values of max_skip

jori-nordic commented:
> Fix the host side not setting sec_adv_max_skip

Do we have a test for that?

Thalley commented:
> > Fix the host side not setting sec_adv_max_skip
> 
> Do we have a test for that?

Clearly not, since we've never set the value. Not sure if that counts as a bug or a missing enhancement :D

Thalley commented on subsys/bluetooth/host/adv.c:
```diff
@@ -1216,6 +1216,8 @@ static int le_ext_adv_param_set(struct bt_le_ext_adv *adv,
 
 	cp->sid = param->sid;
 
+	cp->sec_adv_max_skip = param->secondary_max_skip;
+
```
Suggest to put this change into a separate commit

Tronil replied:
Done

Tronil committed:
Bluetooth: Fix the host side not setting sec_adv_max_skip

Set sec_adv_max_skip

Signed-off-by: Troels Nilsson <trnn@demant.com>

Tronil committed:
Bluetooth: Controller: Implement Secondary_Advertising_Max_Skip

Behaviour is unchanged for max_skip == 0

For max_skip > 0, use a slightly different calculation to ensure
we can pass LL/DDI/ADV/BV-28-C

Signed-off-by: Troels Nilsson <trnn@demant.com>

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_adv.c:
```diff
@@ -1487,20 +1488,31 @@ uint8_t ll_adv_enable(uint8_t enable)
+				 * add ULL_ADV_RANDOM_DELAY and round up for a
+				 * aux interval equal or higher instead
+				 */
+				aux->interval =
+					DIV_ROUND_UP(((uint64_t)adv->interval *
+							  ADV_INT_UNIT_US) +
+							 HAL_TICKER_TICKS_TO_US(
+								ULL_ADV_RANDOM_DELAY),
+							 PERIODIC_INT_UNIT_US);
+			} else {
+				aux->interval =
+					(((uint64_t)adv->interval * ADV_INT_UNIT_US) *
+					 (adv->max_skip + 1)) / PERIODIC_INT_UNIT_US;
```
Suggest to put `(uint64_t)adv->interval * ADV_INT_UNIT_US)` into a variable to make this more readable. The variable can be used both places. 

Similarly the `/ PERIODIC_INT_UNIT_US` should be possible to have common code for too.
